### PR TITLE
Implemented `Hide Stars` animation on cinematics start

### PR DESCRIPTION
**Pull request by moinm3uw**
https://trello.com/c/4nYUdZJ7/775-highbugps-stars-dont-hide-during-cinematic-start

### CODE REVIEW

File: Source/ProgressionSystemRuntime/ProgressionSystemRuntime.Build.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ public ProgressionSystemRuntime(ReadOnlyTargetRules Target) : base(Target)
 				// Bomber modules 
 				"MyUtils" 
 				, "PoolManager" // Star and Widget Actors
-				, "MetaCheatManager" // PSCheatExtension
+				, "MetaCheatManager", "NewMainMenu",  // PSCheatExtension
```
Comment (JanSeliv) [inline Source/ProgressionSystemRuntime/ProgressionSystemRuntime.Build.cs]: Modular Game Feature modules don't allow to be referenced from outside by anything else, even by other similar modules.

Please remove this cross-dependency and see following comment, where I explain how to react on Cinematic start properly

File: Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp [hunk 2]
```diff
@@ -152,6 +157,22 @@ void UPSWorldSubsystem::OnCharacterReady(APlayerCharacter* PlayerCharacter, int3
...
+//  Is called when a main menu state has been changed
+void UPSWorldSubsystem::OnMainMenuStateChanged(ENMMState NewState)
+{
+	if (NewState == ENMMState::Cinematic)
+	{
+		for (AActor* Actor : SpawnedStarActorsInternal)
+		{
+			if (Actor)
+			{
+				Actor->SetActorHiddenInGame(true);
```
Comment (JanSeliv) [inline Source/ProgressionSystemRuntime/Private/Data/PSWorldSubsystem.cpp]: From Trello card:

> Expected result:
> Once cinematic started, stars smoothly disappeared: each star scales from (1,1,1) scale to (0,0,0) for X seconds (by default 0.2sec)

It means, you can't do `SetActorHiddenInGame` as it will instantly hide starts right in front of player's view that looks as a bug again.

Remove `SetActorHiddenInGame ` usage and see following comment, where I explain how to implement animating the star properly

**Comment by JanSeliv** [comment]
I converted the bug to a task and explained code design. Please read carefully as I spent around 5 hours preparing code and explaining the approach
https://trello.com/c/4nYUdZJ7/775-highbugps-stars-dont-hide-during-cinematic-start

**Comment by JanSeliv** [comment]
Submitting the [Progression Lock Not Reapplied on Client After Returning to Menu](https://github.com/h4rdmol/ProgressionSystem/pull/3) caused conflict to this review as the same places are being changed. Please resolve conflict and make sure it complies

**Comment by moinm3uw** [comment]
Added a new logic for stars disappearing 
https://github.com/user-attachments/assets/b91bcb9d-f471-49ee-bb5b-47215a48c58f

File: Source/ProgressionSystemRuntime/ProgressionSystemRuntime.Build.cs [hunk 3]
```diff
@@ -23,7 +23,7 @@ public ProgressionSystemRuntime(ReadOnlyTargetRules Target) : base(Target)
 				// Bomber modules 
 				"MyUtils" 
 				, "PoolManager" // Star and Widget Actors
-				, "MetaCheatManager" // PSCheatExtension
+				, "MetaCheatManager",  // PSCheatExtension
```
Comment (JanSeliv) [inline Source/ProgressionSystemRuntime/ProgressionSystemRuntime.Build.cs]: Please revert this unnecessary change

File: Source/ProgressionSystemRuntime/Public/Data/PSTypes.h [hunk 4]
```diff
@@ -60,4 +60,8 @@ struct FPSRowData : public FTableRowBase
 	/** Defines if level is locked or not */
 	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category="C++")
 	bool IsLevelLocked = true;
+
+	/** Defines the star animation for each character */
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category="C++")
+	TObjectPtr<class UCurveTable> HideStarsAnimation;
```
Comment (JanSeliv) [inline Source/ProgressionSystemRuntime/Public/Data/PSTypes.h]: inline init the property with `= nullptr;`

File: Source/ProgressionSystemRuntime/Public/Data/PSTypes.h [hunk 5]
```diff
@@ -60,4 +60,8 @@ struct FPSRowData : public FTableRowBase
 	/** Defines if level is locked or not */
 	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category="C++")
 	bool IsLevelLocked = true;
+
+	/** Defines the star animation for each character */
```
Comment (JanSeliv) [inline Source/ProgressionSystemRuntime/Public/Data/PSTypes.h]: There are might be multiple animations in this row, please mention this animation is called on in-game cinematic started

File: Source/ProgressionSystemRuntime/Public/LevelActors/PSStarActor.h [hunk 6]
```diff
@@ -9,21 +9,39 @@ UCLASS()
...
+
 protected:
 	// Called when the game starts or when spawned
 	virtual void BeginPlay() override;
+
+	/** Function called every frame on this Actor */
+	virtual void Tick(float DeltaTime) override;
+	/** Static mesh of actor */
 	UPROPERTY()
 	TObjectPtr<class UStaticMeshComponent> MeshComponentInternal = nullptr;
```
Comment (JanSeliv) [inline Source/ProgressionSystemRuntime/Public/LevelActors/PSStarActor.h]: Please remove `SetStaticMesh` and `MeshComponentInternal` as redundant,as I remember its functionality is fully set in actor now

File: Source/ProgressionSystemRuntime/Public/LevelActors/PSStarActor.h [hunk 7]
```diff
@@ -9,21 +9,39 @@ UCLASS()
...
+
+	/** When a cinematicStarted */
+	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
+	void OnAnyCinematicStarted(const UObject* LevelSequence, const UObject* FromInstigator);
+
+	/** Hiding stars with animation */
+	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
+	void TryPlayHideStarAnimation();
+
+public:
```
Comment (JanSeliv) [inline Source/ProgressionSystemRuntime/Public/LevelActors/PSStarActor.h]: redundant `public` specifier at the end of class

File: Source/ProgressionSystemRuntime/Public/Data/PSTypes.h (hunk 4 above)
Comment (moinm3uw) [reply Source/ProgressionSystemRuntime/Public/Data/PSTypes.h]: done as part of https://github.com/h4rdmol/ProgressionSystem/pull/12/commits/0dc42fbdf6ceb67efd71cbafead115ade2be8ad1

File: Source/ProgressionSystemRuntime/ProgressionSystemRuntime.Build.cs (hunk 3 above)
Comment (moinm3uw) [reply Source/ProgressionSystemRuntime/ProgressionSystemRuntime.Build.cs]: done as part of https://github.com/h4rdmol/ProgressionSystem/pull/12/commits/0dc42fbdf6ceb67efd71cbafead115ade2be8ad1 and https://github.com/h4rdmol/ProgressionSystem/pull/12/commits/205b74a50da8effcf5f18761ff1eb503c2a7abd7

File: Source/ProgressionSystemRuntime/Public/LevelActors/PSStarActor.h (hunk 7 above)
Comment (moinm3uw) [reply Source/ProgressionSystemRuntime/Public/LevelActors/PSStarActor.h]: done as part of https://github.com/h4rdmol/ProgressionSystem/pull/12/commits/82d67ef9c4cfefaf80d1e43ecfa1acc873db5dac

File: Source/ProgressionSystemRuntime/Public/Data/PSTypes.h (hunk 5 above)
Comment (moinm3uw) [reply Source/ProgressionSystemRuntime/Public/Data/PSTypes.h]: updated as part of https://github.com/h4rdmol/ProgressionSystem/pull/12/commits/82d67ef9c4cfefaf80d1e43ecfa1acc873db5dac

File: Source/ProgressionSystemRuntime/Public/LevelActors/PSStarActor.h (hunk 6 above)
Comment (moinm3uw) [reply Source/ProgressionSystemRuntime/Public/LevelActors/PSStarActor.h]: removed as part of https://github.com/h4rdmol/ProgressionSystem/pull/12/commits/5d72f8876997aaa97e4601fe31363fec4893004e

File: Source/ProgressionSystemRuntime/Public/LevelActors/PSStarActor.h [hunk 8]
```diff
@@ -9,21 +9,31 @@ UCLASS()
...
+
+	/** Stores the starting time to hide stars */
+	UPROPERTY(VisibleInstanceOnly, BlueprintReadWrite, Transient, Category = "C++", meta = (BlueprintProtected, DisplayName = "Starting time to hide stars"))
+	float StartTimeHideStarsInternal = 0.0f;
+
+	/** When a local character load finished */
+	UFUNCTION(BlueprintCallable, Category= "C++", meta = (BlueprintProtected))
+	void OnLocalCharacterReady(APlayerCharacter* Character, int32 CharacterID);
+
+	/** When a cinematicStarted */
```
Comment (JanSeliv) [inline Source/ProgressionSystemRuntime/Public/LevelActors/PSStarActor.h]: Wrong description here and in .cpp

Comment (moinm3uw) [reply Source/ProgressionSystemRuntime/Public/LevelActors/PSStarActor.h]: improved comments 
 with https://github.com/h4rdmol/ProgressionSystem/pull/12/commits/7c26ce9bd69bde78932c3b67cf9ef1a332c48922
